drltrace/drltrace_frontend.cpp

```diff
@@ -0,0 +1,314 @@
+
+static void
+print_version() {
+#if defined(BUILD_NUMBER) && defined(VERSION_NUMBER)
+    printf("drltrace version %s -- build %d\n", STRINGIFY(VERSION_NUMBER), BUILD_NUMBER);
+#elif defined(BUILD_NUMBER)
+    printf(TOOLNAME" custom build %d -- %s\n", BUILD_NUMBER, __DATE__);
+#else
+    printf(TOOLNAME" custom build -- %s, %s\n", __DATE__, __TIME__);
+#endif
+    exit(0);
+}
+
+static void
+configure_application(char *app_name, char **app_argv, void **inject_data,
+                      const char *dr_root, const char *lib_path)
+{
+    bool is_debug = false;
+#ifdef DEBUG
+    is_debug = true;
+#endif
+    int errcode;
+    char *process;
+    process_id_t pid;
+    char dr_option[MAX_DR_CMDLINE];
+    dr_option[0] = '\0';
+
+    if (op_no_follow_children.get_value() == true)
+        _snprintf(dr_option, strlen("-no_follow_children") + 1, "-no_follow_children");
+    NULL_TERMINATE_BUFFER(dr_option);
+
+#ifdef UNIX
+    errcode = dr_inject_prepare_to_exec(app_name, (const char **)app_argv, inject_data);
+#else
+    errcode = dr_inject_process_create(app_name, (const char **)app_argv, inject_data);
+#endif
+    if (errcode != 0 && errcode != WARN_IMAGE_MACHINE_TYPE_MISMATCH_EXE) {
+        std::string msg =
+            std::string("failed to create process for \"") + app_name + "\"";
+#ifdef WINDOWS
+        char buf[MAXIMUM_PATH];
+        int sofar = _snprintf(buf, BUFFER_SIZE_ELEMENTS(buf), "%s", msg.c_str());
+        FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                      NULL, errcode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                      (LPTSTR) buf + sofar,
+                      BUFFER_SIZE_ELEMENTS(buf) - sofar*sizeof(char), NULL);
+#endif
+        DRLTRACE_ERROR("%s", msg.c_str());
+    }
+
+    pid = dr_inject_get_process_id(*inject_data);
+
+    process = dr_inject_get_image_name(*inject_data);
+    if (dr_register_process(process, pid,
+                            false, dr_root,
+                            DR_MODE_CODE_MANIPULATION,
+                            is_debug, DR_PLATFORM_DEFAULT,
+                            dr_option) != DR_SUCCESS) {
+        DRLTRACE_ERROR("failed to register DynamoRIO configuration");
+    }
+
+    if (dr_register_client(process, pid, false, DR_PLATFORM_DEFAULT, 0, 0, lib_path,
+                           op_ltracelib_ops.get_value().c_str()) != DR_SUCCESS) {
+        DRLTRACE_ERROR("failed to register DynamoRIO client configuration");
+    }
+}
+
+static void
+check_logdir_path(const char *logdir) {
+    drfront_status_t sc;
+    char absolute_logdir_path[MAXIMUM_PATH];
+    bool result;
+
+    sc = drfront_get_absolute_path(logdir, absolute_logdir_path,
+                                   BUFFER_SIZE_ELEMENTS(absolute_logdir_path));
+    if (sc != DRFRONT_SUCCESS)
+        DRLTRACE_ERROR("drfront_get_absolute_path failed, error code = %d\n", sc);
+
+    if (drfront_access(absolute_logdir_path, DRFRONT_WRITE, &result) != DRFRONT_SUCCESS)
```

Comment:
See comment above: on Android drfront_access() is sometimes not sufficient to detect accessibility and we need the try-write stuff in the appdata_logdir code.  (But also xref i#220 which could change that.)